src/sdk/Audio/SpeakerAudioDestination.ts

```diff
@@ -85,27 +87,40 @@ export class SpeakerAudioDestination implements IAudioDestination, IPlayer {
                 }
             });
         } else if (this.privAudioOutputStream !== undefined) {
-            let receivedAudio = new ArrayBuffer(this.privBytesReceived);
-            this.privAudioOutputStream.read(receivedAudio).then((_: number): void => {
-                if (this.privFormat.hasHeader) {
-                    receivedAudio = SynthesisAdapterBase.addHeader(receivedAudio, this.privFormat);
+            if ((this.privFormat.formatTag === AudioFormatTag.PCM || this.privFormat.formatTag === AudioFormatTag.MuLaw) && this.privFormat.hasHeader === false) {
+                // tslint:disable-next-line:no-console
+                console.warn(`Play back is not supported for raw PCM/mulaw format.`);
+                if (!!this.onAudioEnd) {
+                    this.onAudioEnd(this);
                 }
-                const audioBlob = new Blob([receivedAudio], { type: AudioFormatToMimeType[this.privFormat.formatTag] });
-                this.privAudio.src = window.URL.createObjectURL(audioBlob);
-                this.notifyPlayback().then(() => {
-                    if (!!cb) {
-                        cb();
+            } else {
+                let receivedAudio = new ArrayBuffer(this.privBytesReceived);
+                this.privAudioOutputStream.read(receivedAudio).then((_: number): void => {
+                    if (this.privFormat.hasHeader) {
+                        receivedAudio = SynthesisAdapterBase.addHeader(receivedAudio, this.privFormat);
```

Comment:
Doesn't addHeader already perform the hasHeader() check?

Reply:
The `receivedAudio` is from `privAudioOutputStream` and has no header. So if the format needs a header, we need to add it here and send the audio chunk with header to browser for playback.

Reply:
My comment is about the line:
 `
if (this.privFormat.hasHeader) {`

The first three lines of SynthesisAdapterBase.addHeader are:
```
    public static addHeader(audio: ArrayBuffer, format: AudioOutputFormatImpl): ArrayBuffer {
        if (!format.hasHeader) {
            return audio;
        }
```
Do we need to perform the hasHeader check twice?

Reply:
Yes, you are right, I removed the duplicate checking. Thanks!